src/main/java/gtPlusPlus/core/item/general/ItemMagicFeather.java

```diff
@@ -0,0 +1,269 @@
+			player.capabilities.allowFlying = true;
+		}
+		player.sendPlayerAbilities();
+	}
+
+	@SubscribeEvent(priority=EventPriority.HIGHEST)
+	public void onPlayerTick(TickEvent.PlayerTickEvent event) {
+		if (event.side != Side.SERVER) {
+			return;
+		}
+		EntityPlayer player = event.player;
+		HashSet<TileEntityBeacon> aBeaconData = sBeaconData.get(player.getUniqueID());
+		if (aBeaconData != null && !aBeaconData.isEmpty()) {
+			HashSet<TileEntityBeacon> aRemovals = new HashSet<TileEntityBeacon>();
+			for (TileEntityBeacon aBeacon : aBeaconData) {
+				int level = aBeacon.getLevels();
+				if (level == 0) {
+					aRemovals.add(aBeacon);
+					continue;
+				}
+				int radius = (level * 10 + 10);
+				int x = aBeacon.xCoord;
+				int z = aBeacon.zCoord;
+				if (player.posX < (x - radius) || player.posX > (x + radius)) {
+					aRemovals.add(aBeacon);
+					continue;
+				}
+				if (player.posZ < (z - radius) || player.posZ > (z + radius)) {
+					aRemovals.add(aBeacon);
+					continue;
+				}
+			}
+			aBeaconData.removeAll(aRemovals);
+		}
+		boolean hasItem = hasItem(player, ModItems.itemMagicFeather);
+		if (!hasItem) {
+			ItemMagicFeather.sPlayerData.remove(player.getUniqueID());			
+		}		
+		MagicFeatherData data = ItemMagicFeather.sPlayerData.get(player.getUniqueID());
+		if (data == null) {
+			data = new MagicFeatherData(player);
+			ItemMagicFeather.sPlayerData.put(player.getUniqueID(), data);
+		}
+		data.onTick();
+	}
+	
+	@SubscribeEvent(priority=EventPriority.LOWEST)
+	public void onPlayerDeath(LivingDeathEvent event) {		
+		if (event.entityLiving != null) {
+			EntityLivingBase aEntity = event.entityLiving;
+			if (aEntity instanceof EntityPlayer && aEntity.worldObj != null) {
+				if (aEntity.worldObj.isRemote) {
+					return;
+				}
+				else {
+					EntityPlayer aPlayer = (EntityPlayer) aEntity;
+					ItemMagicFeather.sPlayerData.remove(aPlayer.getUniqueID());
+					ItemMagicFeather.sBeaconData.remove(aPlayer.getUniqueID());
+				}
+			}			
+		}		
+	}
+
+	private static boolean hasItem(EntityPlayer player, Item item) {
+		for (int i = 0; i < player.inventory.getSizeInventory(); i++) {
+			ItemStack stack = player.inventory.getStackInSlot(i);
+			if (stack != null && stack.getItem() != null && stack.getItem() instanceof ItemMagicFeather) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	private static class MagicFeatherData {
+
+		private final EntityPlayer player;
+		private boolean hasItem = false;
+		private int checkTick = 0;
+		private boolean beaconInRangeCache;
+
+		public MagicFeatherData(EntityPlayer player) {
+			this.player = player;
+			this.beaconInRangeCache = player.capabilities.allowFlying;
+		}
+
+		public void onTick() {
+			try {
+			boolean hasItem = hasItem(player, ModItems.itemMagicFeather);
+			if (hasItem != this.hasItem) {
+				if (hasItem) {
+					this.onAdd();
+				}
+				if (!hasItem) {
+					this.onRemove();
+				}
+				this.hasItem = hasItem;
+				Logger.INFO("Ticking feather "+hasItem);
```

Comment:
A2

Reply:
GT++'s logger doesn't log anything unless you explicitly turn it on.

Reply:
Heh, mark it as changes requested or dream will beat me to removing this.